### ISSUE TITLE
Proper subtraction image S errors with default Sampling parameter of imProc.transIm.interp2wcs - possible implications for co-addition

### ISSUE THREAD

**Issue by RuslanKonno** [EranOfek/AstroPack#377]
The error appears on the following example of images as saved on euler:

Reference image: /raid/eran/projects/telescopes/LAST/Images_PipeTest/testPipe/LAST.01.02.02/2023/04/25/proc/210027v0/LAST.01.02.01_20230425.210027.515_clear_185-02_001_001_016_sci_proc_Image_1.fits

New1 image: /raid/eran/projects/telescopes/LAST/Images_PipeTest/testPipe/LAST.01.02.02/2023/04/25/proc/211834v0/LAST.01.02.01_20230425.211934.870_clear_185-02_004_001_016_sci_proc_Image_1.fits

New2 image: /raid/eran/projects/telescopes/LAST/Images_PipeTest/testPipe/LAST.01.02.02/2023/04/25/proc/212514v0/LAST.01.02.01_20230425.212915.038_clear_185-02_013_001_016_sci_proc_Image_1.fits

The subtraction was performed using the following script: 
[script.zip](https://github.com/EranOfek/AstroPack/files/13950994/script.zip)

When I substract the reference image from the New1 image and plot D and S, the S score is positive around the holes in D and negative around the peaks in D:

![Screenshot from 2024-01-16 15-49-58](https://github.com/EranOfek/AstroPack/assets/51087294/48813061-e40e-4f34-878d-7a0761a4a389)

This does not happen in the example of New2:

![Screenshot from 2024-01-16 15-54-45](https://github.com/EranOfek/AstroPack/assets/51087294/b769dc7f-36ff-498c-80ac-0c9a9bb69364)

Both of the examples above are produced with the default Sampling parameter of imProc.transIm.interp2wcs of 20 using the cubic interpolation. If I set the Sampling parameter to 1, this effect no longer occurs:

New1: 
![Screenshot from 2024-01-16 15-58-28](https://github.com/EranOfek/AstroPack/assets/51087294/61172fae-3907-452c-a2eb-1aa93bee6f2c)

New2:
![Screenshot from 2024-01-16 16-00-16](https://github.com/EranOfek/AstroPack/assets/51087294/296557c7-3d37-4996-9cb8-c40ef277acc6)

The issue seems to improve with a finer registration grid. It is strange that S is influenced while D is not - this will require further investigation. 

Furthermore, the same registration parameters are used for the co-addition - so it's an open question if the registration grid sampling may also be causing issues in the co-addition.

**Comment by RuslanKonno** [comment]
The PSF of D for New1 does not seem to change. 

![Screenshot from 2024-01-16 16-21-37](https://github.com/EranOfek/AstroPack/assets/51087294/e648c9e6-ff2b-4e80-9690-7e177291a865)

**Comment by RuslanKonno** [comment]
Sum and median of S (after IFFT, before normalization) changes, while std does not:

Sampling = 1:   median = 0.0014602, sum = -8375.7, std = 0.23431.
Sampling = 20: median = -0.0002674, sum = -13309, std = 0.23411.

Number of NaN values also changes:
Num NaN in Sampling = 1: 87444
Num NaN in Sampling = 20: 95979

-> ~10% more NaN pixels

**Comment by RuslanKonno** [comment]
Similar results for D -> D does indeed change

Sampling = 1: median = 0.0003819, sum = -8224.7, std = 1.0007.
Sampling = 20: median = 0.0020676, sum = -13204, std = 1.0002.

Number of NaN values:
Num NaN in Sampling = 1: 87444
Num NaN in Sampling = 20: 95979

-> ~10% more NaN pixels

**Comment by RuslanKonno** [comment]
Doing the same for the New1 image post registration.

Sampling = 1: median = 195.5, sum = 574185664, std = 11.541.
Sampling = 20: median = 195.51, sum = 572519232, std = 11.538.

Number of NaN values:
Num NaN in Sampling = 1: 87444
Num NaN in Sampling = 20: 95979

-> ~10% more NaN pixels

Seems the NaN pixels are caused exactly by the registration grid -> Issue is independent of subtraction.

**Comment by RuslanKonno** [comment]
The different grid sampling leads to clipping of the registered image, leading to further differences down the line, possibly in particular in Fourier space.

![Screenshot from 2024-01-16 17-16-42](https://github.com/EranOfek/AstroPack/assets/51087294/2a19c328-1f24-4297-a437-6cc7023ff405)

**Comment by RuslanKonno** [comment]
The flip of S occured since the NaN values in the new image after registration were replaced with zeroes during the subtraction process. Sources in the reference image within the zero border in the new image led to holes in the difference image. While this border can be masked, the subtraction is still carried out in Fourier space and thus the zero border does effect the entire image. The median of the S image is then lowered to below zero and since we normalized with the median, we multiply with a negative number and thus flip the image in sign.

Given the new approach within AstroDiff to replace NaN values with the mean background value after registration instead of zero, the sign flip is no longer reproduced.
